### ISSUE TITLE
GPL-2.1[sic] license generates invalid SPDX RDF file

### ISSUE THREAD

**Issue by timurphy** [fossology/fossology#643]
Tested on version `3.0.0-165-g27dea95`.

Uploading a file which matches the `GPL-2.1[sic]` license results in an invalid SPDX RDF file, as the square bracket is not a valid URI character.

Steps to reproduce:

1. Upload from URL https://gist.githubusercontent.com/timurphy/56eb73da5992bf4423af/raw/6273b0b41d25f741a53b2bff52626b1d4f56a698/lgpl_sic.txt with Nomos scan enabled
2. Generate SPDX RDF file, which will look like this: https://gist.github.com/timurphy/25f0b72b2b93308f68b7#file-lgpl_sic_spdx-rdf
3. Using the [spdx tools](https://github.com/spdx/tools) project, run the Verify command, which will print this error:

```
 WARN [main] (RDFDefaultErrorHandler.java:36) - file:////tmp/lgpl_spdx.rdf(line 21 column 137): {W107} Bad URI: <file:///srv/fossology/repository/report/SPDX2_lgpl_sic.txt_1456357178.rdf#LicenseRef-GPL-2.1[sic]> Code: 0/ILLEGAL_CHARACTER in FRAGMENT: The character violates the grammar rules for URIs/IRIs.
 WARN [main] (RDFDefaultErrorHandler.java:36) - file:////tmp/lgpl_spdx.rdf(line 76 column 149): {W107} Bad URI: <file:///srv/fossology/repository/report/SPDX2_lgpl_sic.txt_1456357178.rdf#LicenseRef-GPL-2.1[sic]> Code: 0/ILLEGAL_CHARACTER in FRAGMENT: The character violates the grammar rules for URIs/IRIs.
This SPDX Document is not valid due to:
	Invalid license id 'LicenseRef-GPL-2.1[sic]'.  Must start with 'LicenseRef-' and made up of the characters from the set 'a'-'z', 'A'-'Z', '0'-'9', '+', '_', '.', and '-'.
```

Alternatively, you can validate the RDF file using [this online tool](https://www.w3.org/RDF/Validator/) which gives the same errors.

**Comment by maximilianhuber** [comment]
Good point. I will have a look how to fix this

**Comment by maximilianhuber** [comment]
This issue will be closed with PR #633
